kubeflow/apps/pipeline/Kptfile

```diff
@@ -26,3 +28,153 @@ openAPI:
+            ref: '#/definitions/io.k8s.cli.setters.gcloud.core.project'
+    io.k8s.cli.substitutions.bucket:
+      x-k8s-cli:
+        substitution:
+          name: bucket
+          pattern: ${gcloud.core.project}-kfp-artifacts
+          values:
+          - marker: ${gcloud.core.project}
+            ref: '#/definitions/io.k8s.cli.setters.gcloud.core.project'
+    io.k8s.cli.substitutions.kfp-gcs-gateway-ksa-member:
+      x-k8s-cli:
+        substitution:
+          name: kfp-gcs-gateway-ksa-member
+          pattern: serviceAccount:${gcloud.core.project}.svc.id.goog[kubeflow/kubeflow-pipelines-minio-gcs-gateway]
+          values:
+          - marker: ${gcloud.core.project}
+            ref: '#/definitions/io.k8s.cli.setters.gcloud.core.project'
+    io.k8s.cli.substitutions.user-sa-name:
+      x-k8s-cli:
+        substitution:
+          name: user-sa-name
+          pattern: ${name}-user
+          values:
+          - marker: ${name}
+            ref: '#/definitions/io.k8s.cli.setters.name'
+    io.k8s.cli.substitutions.user-sa-entity:
+      x-k8s-cli:
+        substitution:
+          name: user-sa-entity
+          pattern: user-${name}-user@${gcloud.core.project}.iam.gserviceaccount.com
+          values:
+          - marker: ${name}
+            ref: '#/definitions/io.k8s.cli.setters.name'
+          - marker: ${gcloud.core.project}
+            ref: '#/definitions/io.k8s.cli.setters.gcloud.core.project'
+    io.k8s.cli.substitutions.kfp-cloudsql-client:
+      x-k8s-cli:
+        substitution:
+          name: kfp-cloudsql-client
+          pattern: ${name}-kfp-cloudsql-client
+          values:
+          - marker: ${name}
+            ref: '#/definitions/io.k8s.cli.setters.name'
+    io.k8s.cli.substitutions.kfp-cloudsql-proxy-wi-user:
+      x-k8s-cli:
+        substitution:
+          name: kfp-cloudsql-proxy-wi-user
+          pattern: ${name}-kfp-cloudsql-proxy-wi-user
+          values:
+          - marker: ${name}
+            ref: '#/definitions/io.k8s.cli.setters.name'
+    io.k8s.cli.substitutions.cloudsql-user-kfp-root:
+      x-k8s-cli:
+        substitution:
+          name: cloudsql-user-kfp-root
+          pattern: ${name}-kfp-root
+          values:
+          - marker: ${name}
+            ref: '#/definitions/io.k8s.cli.setters.name'
+    io.k8s.cli.substitutions.gcs-acl:
+      x-k8s-cli:
+        substitution:
+          name: gcs-acl
+          pattern: ${name}-kfp-gcs-acl
+          values:
+          - marker: ${name}
+            ref: '#/definitions/io.k8s.cli.setters.name'
+    io.k8s.cli.substitutions.gcs-wi-user:
+      x-k8s-cli:
+        substitution:
+          name: gcs-wi-user
+          pattern: ${name}-kfp-gcs-wi-user
+          values:
+          - marker: ${name}
+            ref: '#/definitions/io.k8s.cli.setters.name'
+    io.k8s.cli.setters.cloudsql-name:
+      x-k8s-cli:
+        setter:
+          name: cloudsql-name
+          value: dev-4-28-kfp-4
```

Comment:
Use pattern and placeholder to define `cloudsql-name`

Reply:
Cloud SQL and bucket have their own setters because that's critical to enable using an external storage.

Reply:
Also, delete and update doesn't work for cloudsql, because after a store is deleted, the name is preserved for 7 days for data recovery

Reply:
No, I'd say they need to manage cloudsql name by themselves. A counter seems too complicated for deployment script. Users might have their own requirements